### PR TITLE
Pshchegolevatykh

### PR DESCRIPTION
**Pull request by PShchegolevatykh**
Add session 15 tasks

### CODE REVIEW

File: sessions/15/pavel_shchegolevatykh/comprehensions.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    ['Январь', 'Февраль', 'Март', 'Апрель', 'Май', 'Июнь', 'Июль', 'Август', 'Сетнябрь', 'Октябрь', 'Ноябрь', 'Декабрь']
+month_days = [31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31]
+
+months_dict = {name: days for (name, days) in zip(month_names, month_days)}
+print(f'Dict via zip function: {months_dict}')
+
+more_than_30_days = {name for name, days in months_dict.items() if days > 30}
+print(f'More than 30 days months: {more_than_30_days}')
+
+odd_months = [(name, days) for idx, (name, days) in enumerate(months_dict.items(), 1) if idx % 2 == 0]
```
Comment (efedorov-dart) [inline sessions/15/pavel_shchegolevatykh/comprehensions.py]: 1. correct result but wrong name. odd == uneven
2. you should produce tuple not list
3. no days information required in result. just tuple of months

File: sessions/15/pavel_shchegolevatykh/comprehensions.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+print(f'Odd months: {odd_months}')
+
+even_months_generator = (name for name, days in months_dict.items() if days % 2 != 0)
```
Comment (efedorov-dart) [inline sessions/15/pavel_shchegolevatykh/comprehensions.py]: wrong name. This is not even months, but months having odd number of days.

File: sessions/15/pavel_shchegolevatykh/comprehensions.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
+month_names =\
+    ['Январь', 'Февраль', 'Март', 'Апрель', 'Май', 'Июнь', 'Июль', 'Август', 'Сетнябрь', 'Октябрь', 'Ноябрь', 'Декабрь']
```
Comment (efedorov-dart) [inline sessions/15/pavel_shchegolevatykh/comprehensions.py]: typo

File: sessions/15/pavel_shchegolevatykh/default_dict.py [hunk 4]
```diff
@@ -0,0 +1,31 @@
+class MyDefaultDict(dict):
+
+    def __init__(self, *args, **kwargs):
+        if 'default' in kwargs:
+            self.default = kwargs['default']
```
Comment (efedorov-dart) [inline sessions/15/pavel_shchegolevatykh/default_dict.py]: you can simplify it all by just:
kawrgs.pop('default', None)

File: sessions/15/pavel_shchegolevatykh/comprehensions.py (hunk 1 above)
Comment (PShchegolevatykh) [reply sessions/15/pavel_shchegolevatykh/comprehensions.py]: Fixed all 3 points. Thanks.

File: sessions/15/pavel_shchegolevatykh/comprehensions.py (hunk 2 above)
Comment (PShchegolevatykh) [reply sessions/15/pavel_shchegolevatykh/comprehensions.py]: Fixed, thanks.

File: sessions/15/pavel_shchegolevatykh/comprehensions.py (hunk 3 above)
Comment (PShchegolevatykh) [reply sessions/15/pavel_shchegolevatykh/comprehensions.py]: Fixed, thanks.

File: sessions/15/pavel_shchegolevatykh/default_dict.py (hunk 4 above)
Comment (PShchegolevatykh) [reply sessions/15/pavel_shchegolevatykh/default_dict.py]: Very good to know. Thanks! Fixed.

**Pull request merged**